instrumentation/opentelemetry_oban/lib/opentelemetry_oban/job_handler.ex

```diff
@@ -0,0 +1,101 @@
+defmodule OpentelemetryOban.JobHandler do
+  alias OpenTelemetry.Span
+
+  @tracer_id :opentelemetry_oban
+
+  def attach() do
+    attach_job_start_handler()
+    attach_job_stop_handler()
+    attach_job_exception_handler()
+  end
+
+  defp attach_job_start_handler() do
+    :telemetry.attach(
+      "#{__MODULE__}.job_start",
+      [:oban, :job, :start],
+      &__MODULE__.handle_job_start/4,
+      []
+    )
+  end
+
+  defp attach_job_stop_handler() do
+    :telemetry.attach(
+      "#{__MODULE__}.job_stop",
+      [:oban, :job, :stop],
+      &__MODULE__.handle_job_stop/4,
+      []
+    )
+  end
+
+  defp attach_job_exception_handler() do
+    :telemetry.attach(
+      "#{__MODULE__}.job_exception",
+      [:oban, :job, :exception],
+      &__MODULE__.handle_job_exception/4,
+      []
+    )
+  end
+
+  def handle_job_start(_event, _measurements, metadata, _config) do
+    %{
+      job: %{
+        id: id,
+        queue: queue,
+        worker: worker,
+        priority: priority,
+        inserted_at: inserted_at,
+        scheduled_at: scheduled_at,
+        attempt: attempt,
+        max_attempts: max_attempts,
+        meta: job_meta
+      }
+    } = metadata
+
+    :otel_propagator_text_map.extract(Map.to_list(job_meta))
+    parent = OpenTelemetry.Tracer.current_span_ctx()
+    links = if parent == :undefined, do: [], else: [OpenTelemetry.link(parent)]
+    OpenTelemetry.Tracer.set_current_span(:undefined)
+
+    attributes = [
+      "messaging.system": "oban",
+      "messaging.destination": queue,
+      "messaging.destination_kind": "queue",
+      "messaging.operation": "process",
+      "messaging.oban.job_id": id,
+      "messaging.oban.worker": worker,
+      "messaging.oban.priority": priority,
+      "messaging.oban.attempt": attempt,
+      "messaging.oban.max_attempts": max_attempts,
+      "messaging.oban.inserted_at":
+        if(inserted_at, do: DateTime.to_iso8601(inserted_at), else: nil),
+      "messaging.oban.scheduled_at": DateTime.to_iso8601(scheduled_at)
+    ]
+
+    span_name = "#{worker} process"
+
+    OpentelemetryTelemetry.start_telemetry_span(@tracer_id, span_name, metadata, %{
+      kind: :consumer,
+      links: links
+    })
+    |> Span.set_attributes(attributes)
```

Comment:
@indrekj why is this done with `Span.set_attributes`? It's not possible to set the attributes directly in the `start_opts` here? The reason why I'm asking is that I don't see the attributes in my sampler (but they do appear in Honeycomb). I'm receiving this span without the attributes in my sampler. @tsloughter is this intended behaviour of the sampler / span design?

Reply:
I think I used some other project as an example. The `start_opts` works with `attributes` key indeed: https://github.com/open-telemetry/opentelemetry-erlang-contrib/compare/9ff9f64044d47306c0764dd9e12472a1b7eb3349..5b50df46ca8c0847f89e595cf74d4a1e2892c6e8 . I'm also using set_attributes in two other places (insert functions) but there I don't have access to all attributes immediately so there it makes sense to use set_attributes.

Reply:
With this change now I do get the attributes in the sampler, but I'm not sure if this is a bug or not, I was expecting this to work also with `Span.set_attributes`.

Reply:
I think it **might** be intentional. Imagine you have this flow:

1. Start span A
2. Send outbound request to service B
3. Set attributes to span A
4. Do something and finish span A

In the step 2, the sampling decision has to be known. Otherwise, service B doesn't know if the spans should be sampled or not. So the sampler has to be run in the step 1.